File: RazorSvelte/App/shared/urls.ts
import { getFromJson } from "./config";

const urls = getFromJson<{
    authorizedUrl: string;
    indexUrl: string;
    errorUrl: string;
    privacyUrl: string;
    loginUrl: string;
    logoutUrl: string;
    aboutUrl: string;
    spaUrl: string;
    signInGoogleUrl: string;
    signInLinkedInUrl: string;
    signInGitHubUrl: string;
    
    chart1Url: string;
    chart2Url: string;
    chart3Url: string;
}>("urls");

export default urls;

File: RazorSvelte/Pages/_Urls.cs
using Newtonsoft.Json;

namespace RazorSvelte.Pages;

public class Urls
{
    [JsonProperty] public const string AuthorizedUrl = "/authorized";
    [JsonProperty] public const string IndexUrl = "/";
    public const string ErrorUrl = "/error";
    [JsonProperty] public const string LoginUrl = "/login";
    [JsonProperty] public const string LogoutUrl = "/logout";
    [JsonProperty] public const string PrivacyUrl = "/privacy";
    [JsonProperty] public const string SpaUrl = "/spa";
    [JsonProperty] public const string OffcanvasNavUrl = "/offcanvas-nav";
    public const string NotFoundUrl = "/404";
    public const string UnathorizedUrl = "/401";
    [JsonProperty] public const string SignInGoogleUrl = "/signin-google";
    [JsonProperty] public const string SignInLinkedInUrl = "/signin-linkedin";
    [JsonProperty] public const string SignInGitHubUrl = "/signin-github";

    public static string Json { get; private set; }

    static Urls()
    {
        Json = JsonConvert.SerializeObject(new Urls());
    }
}
